src/components/DomainSearch/DomainSearch.tsx

```diff
@@ -0,0 +1,139 @@
+import type { FC } from 'react';
+
+import { useState } from 'react';
+import classNames from 'classnames/bind';
+import { Input, Button, LoadingIndicator } from '@zero-tech/zui/components';
+import { DEFAULT_NETWORK_PROTOCAL } from '../../constants/network';
+import { URLS } from '../../constants/urls';
+import { useBuyDomain, useDomainAvailability } from '../../hooks';
+import styles from './DomainSearch.module.scss';
+
+const cx = classNames.bind(styles);
+
+type DomainSearchTypes = 'default' | 'alternative' | 'nobutton';
+
+type DomainSearchProps = {
+	type?: DomainSearchTypes;
+	isLoadingBalance?: boolean;
+	balance?: string;
+	onBuyButtonClick?: (domainName: string) => void;
+};
+
+export const DomainSearch: FC<DomainSearchProps> = ({
+	type = 'default',
+	isLoadingBalance,
+	balance,
+	onBuyButtonClick,
+}) => {
+	const [focused, setFocused] = useState<boolean>(false);
+
+	const { domainName, setDomainName } = useBuyDomain();
+
+	const showStartEnhancer = focused || domainName.length > 0;
+
+	const {
+		isLoading: isCheckingDomainAvailability,
+		isDomainSearchEnabled,
+		isDomainAvailable,
+		domainPrice,
+	} = useDomainAvailability();
+
+	const isDomainPriceExpensive =
```

Comment:
Change variable name to `isInsufficientBalance`.

Reply:
Updated